core/src/main/java/com/oath/oak/Slice.java

```diff
@@ -139,43 +145,29 @@ int getVersion() {
     }
 
     long getMetadataAddress() {
-        return ((DirectBuffer) buffer).address() + offset;
+        return ((DirectBuffer) readBuffer).address() + offset;
```

Comment:
Metadata is also accessed for write... but I assume it is not a problem since it is just an address. But for clearness I would use write buffer

Reply:
done